server/lib/payments.js

```diff
@@ -259,7 +259,7 @@ export const refundPaymentProcessorFeeToCollective = async (transaction, refundT
 export async function createRefundTransaction(transaction, refundedPaymentProcessorFee, data, user) {
   /* If the transaction passed isn't the one from the collective
    * perspective, the opposite transaction is retrieved. */
-  if (transaction.type === DEBIT) {
+  if (transaction.type === DEBIT && transaction.FromCollectiveId !== transaction.CollectiveId) {
```

Comment:
If you add a condition like this, you better add a generous comment to explain why.

Reply:
Modified the existing comment to handle this. Thanks. 👍🏽